unison-cli/src/Unison/CommandLine/OutputMessages.hs

```diff
@@ -2352,6 +2352,11 @@ prettyEntityValidationError = \case
         "",
         P.wrap $ "The error was: " <> P.text err
       ]
+  Share.HashResolutionFailure hash ->
+    P.lines
+      [ P.wrap $ "Failed to resolve data when hashing " <> prettyHash32 hash <> ".",
+        "Please create an issue and report this to the Unison team"
+      ]
```

Comment:
I'd like to change the wording here (and I guess the `InvalidByteEncoding` message, though it doesn't have to be in this PR) to a complete-sentences/friendly/helpful version. I'd suggest an alternative phrasing but I'm not sure exactly what's going on here. What's a `HashResolutionFailure`/`DeclHashResolutionFailure`?

Reply:
To be honest I don't know what a HashResolutionFailure really is, hence the vague messaging. It comes from deeper in the stack. 
https://github.com/unisonweb/unison/blob/trunk/unison-core/src/Unison/Names/ResolutionResult.hs#L16

This just says `ResolutionFailure represents the failure to resolve a given variable.` which isn't terribly detailed. I'm not sure when it's triggered.

Reply:
Ok, no worries.

It looks like it would be triggered during hash verification if hashing failed due to the presence of free vars in the decl, which should never happen post-typechecking anyway, which also makes me wonder if we could restructure that part of the code to skip steps and speed up computing the hashes in this context.

Reply:
I don't think this message will ever appear, so I guess the phrasing isn't important. I went ahead and added a link back to this discussion in a comment, though even that's debatable.